### ISSUE TITLE
Select_features doesn't work for single id

### ISSUE THREAD

**Issue by kr1stjans** [blue-yonder/tsfresh#202]
- Ubuntu 16.04
- v0.7.1

```
n = 1000
test_x = pd.DataFrame()
test_x['id'] = np.ones(n)
test_x['time'] = np.array(range(1, n + 1))
test_x['value'] = np.random.rand(n)
test_y = pd.Series(data=np.random.rand(n))
features_filtered_direct = extract_relevant_features(test_x, test_y, column_id="id", column_sort="time", column_value="value")
```

I am trying to predict a delayed time series data. I have X vector with delayed values and Y vector with actual values. Select_features method fails with X must contain at least two samples. If I use multiple ids it works (like robots examples).

**Comment by MaxBenChrist** [comment]
have you tried it with multiple ids?

**Comment by kr1stjans** [comment]
Yes, it works with multiple ids, but I need only one id. I'm in tsfresh Lobby on Gitter and we can discuss it there.

**Comment by MaxBenChrist** [comment]
can you please post the error stack here?

there was a reason that the calculation of features for one time series causes an error, I just do not remember it right now. I will have to look into that

**Comment by kr1stjans** [comment]
`Feature Extraction: 100%|██████████| 1/1 [00:01<00:00,  1.12s/it]
Traceback (most recent call last):
  File "/home/kristjans/Projects/nedo/prediction.py", line 43, in <module>
    x_values, y_values, dates, columns = data_manager.DataManager().get_data(mm_id)
  File "/home/kristjans/Projects/nedo/data_manager.py", line 100, in get_data
    column_value="value")
  File "/usr/local/lib/python3.5/dist-packages/tsfresh/convenience/relevant_extraction.py", line 53, in extract_relevant_features
    X_sel = select_features(X_ext, y, feature_selection_settings=feature_selection_settings)
  File "/usr/local/lib/python3.5/dist-packages/tsfresh/feature_selection/selection.py", line 94, in select_features
    raise ValueError("X must contain at least two samples.")
ValueError: X must contain at least two samples.`

**Comment by MaxBenChrist** [comment]
Ok we got the issue there, you try to filter features for just one sample. That does not make sense from a statistical point of view. A feature selection for just one sample NEVER makes sense ;)

Just replace extract_relevant_fratures with extract_features and you are fine

**Comment by yashgarg1232** [comment]
@kr1stjans I am running into similar issue for regression problem. Do you happen to have a pseudo code of multiple device?

**Comment by nils-braun** [comment]
What do you mean with "similar problem"? Do you also try to do feature selection with just a single identifier? That is not a problem of tsfresh, but this can conceptionally not work :-)

**Comment by yashgarg1232** [comment]
I am trying to execute select_features between the X and target Y (regression variable). It happens that I am running into the issue, where the system wants Y to be 1D, but my y is 2D (# of devices x # of values per time), as my problem is to forecast. Maybe I am miss understanding. And yes, I have multiple devices and sensors (initially I had only 1 device - test for dummies). My test case, 2 devices, 1 sensor, 1 target sensor for 100 timestamps (real case has many more)

**Comment by nils-braun** [comment]
I think there is some different understanding involved here, yes.

The `select_features` method helps you to select a set of features from your features matrix `X` (a matrix, where each column is a feature and each row is an instance). The `select_features` needs as additional input the target, which tells the function to what it should optimize for. This target `y` needs to have the same number of rows as `X`, because for each of the rows the algorithm needs to know what the correct output (the target) should be.

So having `y` multi-dimension (two-dimensional in your case), does not make sense, because what should the algorithm look for? Relevant to the first column in y? Or the second column?

Maybe you are confusing the feature extraction with the feature selection here? Is `X` maybe your time series?

We have [documentation](https://tsfresh.readthedocs.io/en/latest/text/forecasting.html) available for forecasting and also several [notebooks](https://github.com/blue-yonder/tsfresh/blob/main/notebooks/examples/05%20Timeseries%20Forecasting.ipynb) as well as [helper functions](https://tsfresh.readthedocs.io/en/latest/api/tsfresh.utilities.html#tsfresh.utilities.dataframe_functions.make_forecasting_frame).
In short: what you normally do in forecasting, is to use the data up to point "t" in time, extract the features on only this data (which will give you a single row in your feature matrix) and then use the value of point "t + 1" as the target for this specific row (so you will end up with the same number of rows in `y` in `X`). 

If you are willing to share your data and the exact problem you would like to solve, we might be able to help you!

**Comment by igorkf** [comment]
@nils-braun I don't know why, but I'm having problems with `extract_relevant_features()`. This is the error message:
`AssertionError: X and y must contain the same number of samples.`
    
This is my code:
```
def get_relevant_features(data, y, default_fc_parameters):

    # impute NaN values with mean
    imputer = SimpleImputer()
    imputed_data = pd.DataFrame(imputer.fit_transform(data))
    imputed_data.columns = data.columns
    imputed_data.index = data.index
    print('Imputed NaN for original data')
    assert len(imputed_data) == len(y)
    
    # add index to target
    y.index = imputed_data['date'].tolist()
    print('Added "X" index to "y"')
    
    # extract, impute and select relevant features
    features_filtered_direct = extract_relevant_features(imputed_data,
                                                         y,
                                                         column_id='date',
                                                         column_sort='ts_id',
                                                         default_fc_parameters=default_fc_parameters,
                                                         ml_task='classification')
    
    return features_filtered_direct
```
